File: src/lib/php/DSType.php
<?php

require_once "DSLib.php";
require_once "DSClassInfo.php";

define("DS_TYPE_VERSION", "0.0.1");
define("DS_TYPE_NAME",    "DSType");
define("DS_TYPE_AUTHOR",  "Thorsten Geppert (thorsten@ext2fs.de)");

class DSType extends DSLib {

	public function __construct() {
		parent::__construct();
	}

	public function GetClassInfo() {
		return new DSClassInfo(
			DS_TYPE_NAME,
			DS_TYPE_VERSION,
			DS_TYPE_AUTHOR
		);
	}

}

?>

File: src/lib/cpp/DSType.cpp
#include "DSType.h"

DynSoft::DSType::DSType() : DSLib() {
}

DynSoft::DSClassInfo DynSoft::DSType::GetClassInfo() {
	return DSClassInfo(
		DS_TYPE_NAME,
		DS_TYPE_VERSION,
		DS_TYPE_AUTHOR
	);
}
